### PR TITLE
Extend troubleshooting guide according to findings during matrix support chat

### PR DESCRIPTION
**Pull request by eLtMosen**
There have been multiple instances now where users came into chat and asked for help with a bootloop. Many of them under impression they might have bricked their devices.

I take this as a sign that our current install troubleshoot might profit from some refinement. Especially since the keyword "Bootloop" is not even used as a topic heading currently.

This PR tries to better reassure users and guide them to the appropriate solution steps much more detailed than before.

- Divided guide into boot and flashing problems. Boot problems first due to higher severity. 
- Reordered use of h2, h3 and h4 to better integrate h4 into the new lists.
- Heavily extended all parts according to findings made during support chat on matrix.

grafik
grafik
grafik

### CODE REVIEW

File: templates/includes/install-problems.hbs [hunk 1]
```diff
@@ -2,28 +2,42 @@
...
+      <li><h4>Clean your watch and charger contacts</h4>Even in case the fastboot flash process succeeds with no errors, it can still silently fail due to connection problems caused by dirty contacts. Clean the pins and pads with isopropyl alcohol. Ideally using that old toothbrush you keep around for electronics cleaning or a microfiber cloth. Carefully sanding the contacts (e.g. with a nailfile) can help in case of obvious corrosion.</li>
+      <li><h4>Ensure a stable USB connection</h4>Put the watch and your computer on the same flat surface. Try to not move the cable connecting your computer and the watch during the whole flashing process. In case of worn or corroded contacts and pogo-pins that lost their springiness, it might be useful to attach the watch to its cradle or charging cable using ductape or rubber bands.</li>
+      <li><h4>Repeat the flash process.</h4>Up to five tries have been reported to possibly give a successful result in the end.</li>
```
Comment (dodoradio) [inline templates/includes/install-problems.hbs:14]: `Up to five tries have been reported to possibly give a successful result in the end.`
The phrasing here is a bit strange - I'd try something like:
`it can take up to five flashing attempts for a successful result`
or
`Some users have reported needing up to five flashing attempts for a successful result`
in whatever combination sounds best to you

File: templates/includes/install-problems.hbs [hunk 2]
```diff
@@ -2,28 +2,42 @@
 <div id="InstallTroubleshooting" class="page-header">
   <h1>Installation Troubleshooting</h1>
 </div>
-
+<h2>Boot problems</h2>
 <div class="callout">
-    <h4>Commands fail or stop during the ~5 minute flashing process</h4>
-    Smartwatches are prone to lose a proper connection to their cradle over time due to sweat corroding the watch pins. This is a particularly a regular problem with dory (LG G Watch). Cleaning the pins with isopropyl alcohol or even carefully sanding them (e.g. with a nailfile) can restore a reliable connection.
+    <h3>Bootloop of any kind or stuck boot process</h3>
+    Don't panic. Non of the commands we advised you to use can damage or brick your watch permanently.<br>
```
Comment (dodoradio) [inline templates/includes/install-problems.hbs:8]: * none 😄

File: templates/includes/install-problems.hbs [hunk 3]
```diff
@@ -2,28 +2,42 @@
...
+    You can always go back to the fastboot bootloader menu, by using the <a href="https://asteroidos.org/wiki/useful-commands/#boot-to-fastboot-bootloader-menu">manual finger combos for your watch</a>.
+    <ul>
+      <li><h4>Verify that you used the correct image files</h4>Some devices like the Asus Zenwatch 2, the Fossil Gen4 models and the MTK6580 watches have more than on system image supplied, for different variants with slightly different hardware. Most watches have a graphical fastboot menu that shows the correct codename for the watch. Others need to be identified using the hardware differences noted on the install page.</li>
```
Comment (dodoradio) [inline templates/includes/install-problems.hbs:11]: again, typo on `one system image`.

File: templates/includes/install-problems.hbs [hunk 4]
```diff
@@ -2,28 +2,42 @@
...
+      <li><h4>Repeat the flash process.</h4>Up to five tries have been reported to possibly give a successful result in the end.</li>
+    </ul>
+    <h3>Boot gets stuck even after repeated reflash</h3>
+    Congratulations, you might have found a new issue for us to solve. This should not happen when using stable builds. The nightlies are however advancing quickly and it is not always possible for us to test implications on all watches accordingly. Since the watches install pages link to the nightlies currently, you likely used those. You can try again and <a href="https://release.asteroidos.org/">install a stable build when available</a>.<br>To help us solve the problem, you could try to collect logs. It is possible that the watch does not show a UI but is already accessible using <code>ssh ceres@192.168.2.15</code> or <code>adb shell</code> in the state the boot process stops. In case ssh or ADB connections work, try the following commands to write logs to local .txt files.
```
Comment (dodoradio) [inline templates/includes/install-problems.hbs:17]: For better flow I'd write `While this should not happen with stable builds, the nightlies are advancing ...`
also, I'm not sure what `implications for all watches` is meant to mean here, it doesn't seem very clear. Maybe it's worth simplifying to something like `...advancing quickly, and are not tested on all watches.` ?

File: templates/includes/install-problems.hbs [hunk 5]
```diff
@@ -2,28 +2,42 @@
...
+    <ul>
+      <li><code>ssh root@192.168.2.15 "dmesg" > dmesg-watch.txt</code></li>
+      <li><code>ssh root@192.168.2.15 "journalctl" > journal-watch.txt</code></li>
+      <li><code>adb shell logcat > logcat-watch.txt</code></li>
```
Comment (dodoradio) [inline templates/includes/install-problems.hbs:21]: The commands here for ssh and adb are confusing, since journal and logcat give different info.
on the first boot after flashing, watches will either present an ssh interface (if usb is listed as working) or an adb interface (if usb is listed as broken) - adb and ssh interfaces will never be presented simultaneously.

Also, everything in dmesg should be in journal log anyway. `adb shell logcat` also won't work on some watches since `/system/bin` isn't in path. 
I'd replace the commands with the following:
```
ssh root@192.168.2.15 "journalctl" > journal-watch.txt
ssh root@192.168.2.15 "/system/bin/logcat" > logcat-watch.txt
```
`or, if your watch presents an ADB interface, use the following commands`
```
adb shell journalctl > journal-watch.txt
adb shell "/system/bin/logcat" > logcat-watch.txt
```

File: templates/includes/install-problems.hbs [hunk 6]
```diff
@@ -2,28 +2,42 @@
...
+    You can share those logs in a new <a href="https://github.com/AsteroidOS/meta-smartwatch/issues">Github issue</a> tagged with the watch codename. For support chat, you are invited to <a href="https://matrix.to/#/#Asteroid:matrix.org">join our Matrix channel</a>.
+    <h3>Watch only boots when powered via USB, but shuts off when booted while worn</h3>
+    This is a clear sign of a worn battery that will soon fail. Maybe you should replace it. The boot process of AsteroidOS uses all cores of a watch to ensure a fast startup. This leads to power spikes during the boot process, which can completely shut down watches with worn out batteries. It has been observed that WearOS can still boot on watches with worn out batteries, but AsteroidOS cannot, depending on the progress of battery wear. However, AsteroidOS can usually still provide a useful 10-20 hours of uptime when the watch is worn once booted while powered up. Watch batteries are relatively easy to replace on watches with screwed down backs. Glued back plates usually require a more or less unruly process of heating and prying.
```
Comment (dodoradio) [inline templates/includes/install-problems.hbs:26]: `more or less unruly` is strange. I'm not entirely sure what should go here. maybe words like `frustrating process`, `compromised water resistance` and `creative solutions involving hair driers` should be involved, but I'm not sure how much info should go here

File: templates/includes/install-problems.hbs [hunk 7]
```diff
@@ -2,28 +2,42 @@
...
+    <h3>Watch is not detected and all fastboot commands fail</h3>
+    <ul>
+      <li><h4>When using Linux,</h4> try to prepend <code>sudo</code> to the commands to execute them with root rights. If that works, you are missing UDEV rules for Android devices that would allow your user to issue the commands. Many package managers offer the <code>android-udev-rules</code> package to make the manual entry of those udev rules unnecessary.</li>
+      <li><h4>On Windows systems,</h4> this usually happens due to driver problems. If you have any other Android USB driver installed than the official Google USB driver we link above in the install boxes, please remove that one. Then do a clean install of the Google USB driver or update it by installing a possibly newer version.<br>If that does not solve your problem, we highly recommend to not waste time trying to debug the issue on windows. But create a persistent live Linux USB-Stick according to <a href="https://askubuntu.com/questions/1300454/easy-full-install-usb-that-boots-both-bios-and-uefi">e.g. these instructions at askubuntu.com</a>. And then follow the Linux commands above while booted into the live Linux USB-Stick.</li>
```
Comment (dodoradio) [inline templates/includes/install-problems.hbs:35]: `we highly recommend to not waste time` might be a bit aggressive? I think something like `We find that the linux fastboot drivers are more reliable. It is very easy to create a bootable USB....`

` ...windows. But create ...`
this isn't entirely grammatically correct. ` we highly recommend to not waste time ... on windows, but to create a bootable Linux USB stick...` should be more correct. similarly with `... askubuntu. And then` - this should be `follow these instructions at askubuntu, and then to follow the Linux install commands`

**Review by dodoradio — CHANGES_REQUESTED** [review]
I've been nitpicky about some small grammar things, but also about some larger presentation things, I hope this is appreciated. There are a few more grammar bugs I've noticed, but I'll pick these out once a few more people give their opinions on the overall text.

**Comment by eLtMosen** [comment]
Thank you very much for the review @dodoradio 
I have addressed all of your suggestions and resolved the obvious conversations. Left the two with multiple options unresolved for others to chime in.

File: templates/includes/install-problems.hbs [hunk 8]
```diff
@@ -2,28 +2,50 @@
...
+      <li><h4>Clean your watch and charger contacts</h4>Even in case the fastboot flash process succeeds with no errors, it can still silently fail due to connection problems caused by dirty contacts. Clean the pins and pads with isopropyl alcohol. Ideally using that old toothbrush you keep around for electronics cleaning or a microfiber cloth. Carefully sanding the contacts (e.g. with a nailfile) can help in case of obvious corrosion.</li>
+      <li><h4>Ensure a stable USB connection</h4>Put the watch and your computer on the same flat surface. Try to not move the cable connecting your computer and the watch during the whole flashing process. In case of worn or corroded contacts and pogo-pins that lost their springiness, it might be useful to attach the watch to its cradle or charging cable using ductape or rubber bands.</li>
```
Comment (beroset) [inline templates/includes/install-problems.hbs:13]: "duct tape" is two words.

**Review by beroset — CHANGES_REQUESTED** [review]
Good contribution!  Thanks.

**Review by beroset — APPROVED** [review]

File: templates/includes/install-problems.hbs (hunk 8 above)
Comment (eLtMosen) [reply templates/includes/install-problems.hbs]: Thanks for going through all this text! Typo corrected.

File: templates/includes/install-problems.hbs [hunk 9]
```diff
@@ -2,28 +2,50 @@
...
+      <li><h4>Ensure a stable USB connection</h4>Put the watch and your computer on the same flat surface. Try to not move the cable connecting your computer and the watch during the whole flashing process. In case of worn or corroded contacts and pogo-pins that lost their springiness, it might be useful to attach the watch to its cradle or charging cable using duct tape or rubber bands.</li>
+      <li><h4>Repeat the flash process.</h4>Some users have reported needing up to five flashing attempts for a successful result.<br>There is a small chance that the image files were corrupted during download. We do not currently provide checksums for the files. However, if you download the files again under a different filename and then compare them with <code>sha256sum filename</code>, the checksums must match.</li>
```
Comment (MagneFire) [inline templates/includes/install-problems.hbs:14]: Inconsistent use of `.`'s
image

It's probably better to remove the dot here.

File: templates/includes/install-problems.hbs [hunk 10]
```diff
@@ -2,28 +2,50 @@
...
+    </ul>
+    <h3>Boot gets stuck even after repeated reflash</h3>
+    Congratulations, you might have found a new issue for us to solve. While this should not happen with stable builds, the nightlies are advancing quickly, and are not tested on all watches. Since the watches install pages link to the nightlies currently, you likely used those. You can try again and <a href="https://release.asteroidos.org/">install a stable build when available</a>.<br>To help us solve the problem, you could try to collect logs. It is possible that the watch does not show a UI but is already accessible using <code>ssh ceres@192.168.2.15</code> or <code>adb shell</code> in the state the boot process stops. In case SSH connection works, try the following commands to write logs to local .txt files.
```
Comment (MagneFire) [inline templates/includes/install-problems.hbs:17]: I would rephrase `Since the watches install pages link to the nightlies currently, you likely used those.`, it sounds a bit strange to me.

Maybe something like:
```
You're likely using a nightly build since the install pages link to those currently.
```

File: templates/includes/install-problems.hbs [hunk 11]
```diff
@@ -2,28 +2,50 @@
...
-    <h4>My fastboot command crashes or hangs at "invalid sparse file format at header magi" (with a missing "c" in magic)</h4>
-    This error occurs when using deprecated fastboot and ADB commands on Windows systems. If you already had those commands installed and skipped downloading our <a href="https://release.asteroidos.org/tools/adb_1_0_39+fastboot+ext2simg.zip">supplied zip file</a>, please install those commands instead.
-</div>
-
-<div class="callout">
-    <h4>My Asus Zenwatch 2 flashed successfully but does not boot</h4>
-    There are two versions of the Asus Zenwatch 2, codenamed Sparrow and Wren. Their images are not interchangeable. Please ensure you are using the right files.
+    <h3>Watch is not detected and all fastboot commands fail</h3>
+    <ul>
+      <li><h4>When using Linux,</h4> try prepending <code>sudo</code> to the commands to execute them with root privileges. If that works, you are missing UDEV rules for Android devices that would allow your user to execute the commands. Many package managers provide the <code>android-udev-rules</code>, which avoids the need to manually set up these UDEV rules.</li>
```
Comment (MagneFire) [inline templates/includes/install-problems.hbs:38]: Maybe also provide a direct link in case a package isn't available? 
(https://github.com/M0Rf30/android-udev-rules)

**Review by MagneFire — COMMENTED** [review]
Again, great work!

Just some minor enhancement suggestions from my side, otherwise looking good to me.

File: templates/includes/install-problems.hbs (hunk 9 above)
[the remark this answers is not in the document]
Comment (eLtMosen) [reply templates/includes/install-problems.hbs:14]: That is a great surprise! Coool, i thought it would be complicated to add checksums so did not ask but came up with the re-download work around.
Let's keep the `We do not currently...` sentence until the checksums are actually shown on the website?
Making a new issue for that

Comment (eLtMosen) [reply templates/includes/install-problems.hbs:14]: Yepp, left over from having all sentences end with a dot at some point.

File: templates/includes/install-problems.hbs (hunk 10 above)
Comment (eLtMosen) [reply templates/includes/install-problems.hbs:17]: Great streamlining. Flow is much better like that.

Comment (eLtMosen) [reply templates/includes/install-problems.hbs:17]: opted for `You are likely using a nightly build, as the installation pages currently link to it.`
After some back and forth translation in deepl.com

File: templates/includes/install-problems.hbs (hunk 11 above)
Comment (eLtMosen) [reply templates/includes/install-problems.hbs:38]: Greta idea!

Comment (eLtMosen) [reply templates/includes/install-problems.hbs]: is this okay?
`If the package is not available on your system, you can follow <a href="https://github.com/M0Rf30/android-udev-rules">the instructions at the official android-udev-rules repository</a>.`

Comment (MagneFire) [reply templates/includes/install-problems.hbs]: Sounds good :+1:

**Review by FlorentRevest — COMMENTED** [review]
This is so cool ! Thank you so much @eLtMosen and everybody who contributed to the reviews :D Our users will be very thankful

**Comment by FlorentRevest** [comment]
Oops, I should have squashed that git log but was too excited to merge it... Weellll.... now it's there uh
